src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -291,31 +292,166 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(PhysicalLocation v2Ph
             return physicalLocation;
         }
 
-        internal RegionVersionOne CreateRegion(Region v2Region)
+        internal RegionVersionOne CreateRegion(Region v2Region, Uri uri)
         {
             RegionVersionOne region = null;
 
             if (v2Region != null && (v2Region.StartColumn > 0 ||
                                      v2Region.StartLine > 0 ||
                                      v2Region.EndColumn > 0 ||
                                      v2Region.EndLine > 0 ||
-                                     v2Region.Length > 0 ||
-                                     v2Region.Offset > 0))
+                                     v2Region.ByteLength > 0 ||
+                                     v2Region.ByteOffset > 0 ||
+                                     v2Region.CharLength > 0 ||
+                                     v2Region.CharOffset > 0))
             {
                 region = new RegionVersionOne
                 {
+                    Length = v2Region.ByteLength,
+                    Offset = v2Region.ByteOffset,
                     EndColumn = v2Region.EndColumn,
                     EndLine = v2Region.EndLine,
-                    Length = v2Region.Length,
-                    Offset = v2Region.Offset,
-                    StartColumn = v2Region.StartColumn,
+                    StartColumn = v2Region.StartColumn > 0 ? v2Region.StartColumn : 1,
                     StartLine = v2Region.StartLine
                 };
+
+                if ((v2Region.CharLength > 0 || v2Region.CharOffset > 0) ||
```

Comment:
Again, you can't make any decision based on the value of `CharOffset`. It can validly be 0.

Reply:
Fixed

Reply:
I'm going to need to think about this. I think there are valid cases that are not covered. I'll get back to you...

Reply:
I sent mail about this. Also, I've looked at the logic below now, and it's not quite right. The problem is that you treat the char/offset case and the line/column case separately. That is, the logic embodies the notion that a text region is specified _either_ by charOffset/charLength _or_ by a combination of line/column properties. But the spec says:

> The start of a text region **MAY** be specified by a combination of `startLine` and `startColumn`, or by `charOffset`, or both.
>
> The end of a text region **MAY** be specified by a combination of `endLine` and `endColumn`, or by `charLength`, or both.

So you might specify the start by `charOffset` and the end by `endLine`, `endColumn`. Or you might specify the start by `startLine`, `startColumn` and the end by `charLength`. Or any combination.

Reply:
Yeah this is why I was asking for advice on the conditional flow the other day. I need to meditate...